src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -439,23 +448,23 @@ private MemoryStream GetUncompressedData()
                 }
 
                 // if they start modifying it and the compression method is not "store", we should make sure it will get deflated
-                if (CompressionMethod != CompressionMethodValues.Stored)
+                if (StoredCompressionMethodInternal != ZipCompressionMethod.Stored)
                 {
-                    CompressionMethod = CompressionMethodValues.Deflate;
+                    StoredCompressionMethodInternal = ZipCompressionMethod.Deflate;
                 }
             }
 
             return _storedUncompressedData;
         }
 
-        private CompressionMethodValues CompressionMethod
+        private ZipCompressionMethod StoredCompressionMethodInternal
         {
             get { return _storedCompressionMethod; }
             set
```

Comment:
This can be a public property with a private setter.

```suggestion
        public ZipCompressionMethod CompressionMethod
{
get { return _storedCompressionMethod; }
private set
```